importer/tools/export.hs

```diff
@@ -84,8 +89,8 @@ main = do
     let things = map T.pack $ match !! 0 : otherThings
 
     let (handler, objectSink) = if ".tar" `isSuffixOf` out_path
-            then (\e -> print e >> whenM (doesFileExist out_path) (removeFile out_path), CS.objectToTarEntry .| Tar.tarSink out_path)
-            else (print, Directory.directorySink out_path)
+            then (\e -> printAndDie e >> whenM (doesFileExist out_path) (removeFile out_path), CS.objectToTarEntry .| Tar.tarSink out_path)
+            else (printAndDie, Directory.directorySink out_path)
```

Comment:
Running printAndDie before the other actions (the tar file cleanup) means that the program will exit before the cleanup runs. Maybe move the "andDie" part to the error handler. The last line of main could be something like `whenLeft result (handler >> exitFailure)`